src/Subscriber.js

```diff
@@ -191,6 +193,21 @@ export default class Subscriber {
             throw new Error('unsubscribe: please give a Subscription object as an argument!')
         }
 
+        if (sub.getState() === Subscription.State.unsubscribed) {
+            return Promise.resolve()
+        }
+
+        return Promise.all([
```

Comment:
Out of curiosity, how does this work if we are in the middle of awaiting method `subscribe` above?
```js
await Promise.all([
            sub.waitForSubscribed(),
            this._resendAndSubscribe(sub),
        ])
```